commit 871ccd0182ab925d3d3a7665a5a9d477213371e0 (refs/changes/24/1524/5)
Author: YangYong3 <yy1111.yang@samsung.com>
Date:   2017-03-07T09:43:43+09:00 (2 years, 7 months ago)
    
    joyent/node-triton#172 how about check the current state before change the instance state...

diff --git a/lib/do_instance/gen_do_ACTION.js b/lib/do_instance/gen_do_ACTION.js
index 030c241..28565bf 100644
--- a/lib/do_instance/gen_do_ACTION.js
+++ b/lib/do_instance/gen_do_ACTION.js
@@ -135,10 +135,13 @@ function _doOnEachInstance(self, action, command, state, instances,
                     }
                     alias = arg;
                     uuid = inst.id;
+                    if (state === inst.state) {
+                        console.log('current state is already ' + state);
+                        cb(new Error());
+                    }
                     done();
                 });
             }
-
             // called when "uuid" is set
             function done() {
                 var cOpts = uuid;
